internal/pkg/service/buffer/sink/tablesink/storage/repository/file_repo.go

```diff
@@ -94,10 +94,7 @@ func (r *FileRepository) Get(fileKey storage.FileKey) op.WithResult[storage.File
 //   - Files rotation is done atomically.
 //   - This method is used to rotate files when the import conditions are met.
 func (r *FileRepository) Rotate(rb rollback.Builder, now time.Time, sinkKey key.SinkKey) *op.AtomicOp[storage.File] {
-	var sink definition.Sink
-	return r.
-		rotate(rb, now, sinkKey, &sink, true).
-		ReadOp(r.all.sink.Get(sinkKey).WithResultTo(&sink))
+	return r.rotate(rb, now, sinkKey, nil, true)
 }
```

Comment:
We can replace the `ReadOp` via a `nil` value, and used already present logic in the `rotateAllIn` method.